src/sessionsConverter.js

```diff
@@ -0,0 +1,114 @@
+// module pattern
+var sessionsConverter = (function(){
+  // variable to keep track of different labels
+  var labels = [];
+  // labels set for checking if label is unique
+  var labelSet = {};
+
+  function init(){
+    labels = [];
+    labelSet = {};
+  }
+
+  function getConvertedSessions(sessions){
+    var convertedSessions = [];
+    sessions.forEach(function(session){
+      convertedSessions.push(getConvertedSession(session));
+    });
+    return convertedSessions;
+  }
+
+  function getConvertedSession(session){
+    var now = moment(); // current time
+
+    var outcomes = session.outcomes;
+    // data set item for session
+    var convertedSession = {};
+
+    var momentTimestamp = moment(session.timestamp);
+    var humanisedTimestamp = moment.duration(momentTimestamp.diff(now)).humanize(true);
+
+    convertedSession.label = humanisedTimestamp;
+
+    convertedSession.valuesMap = getValuesMapsFromOutcomes(outcomes);
+
+    // extract data and notes arrays
+    var extractedDataAndNotes = getExtractedDataAndTooltipNotes(convertedSession.valuesMap);
+    convertedSession.data = extractedDataAndNotes.data;
+    convertedSession.notes = extractedDataAndNotes.notes;
+
+
+    var color = Chart.helpers.color;
+    //assign random colour to chart
+    var colour = randomColor({
+      format: 'rgba'
+    });
+    convertedSession.backgroundColor = color(colour).alpha(0.2).rgbString();
+    convertedSession.borderColor = colour;
+    convertedSession.pointBackgroundColor = colour;
+    
+    return convertedSession;
+  }
+
+  function getValuesMapsFromOutcomes(outcomes){
+    var dataMap = {};
+    var noteMap = {};
+    // add each outcome to a set
+    outcomes.forEach(function(outcome){
+      var lowerCaseLabel = outcome.outcome.toLowerCase();
+      updateLabelSet(outcome.outcome);
```

Comment:
re: removing the global variables, one possibility to consider is returning the `outcome.outcome` from this function and building it into the `convertedSession` somewhere. 
Later in  `getChartJSConvertedData` I would perform the required conversions to extract the data formats needed for constructing the `datasets` and `labels` structures. 
So, in summary, the first step would be to gather all the data into a single structure. The second step would be to extract (and potentially transform) the required data for `datasets` and `labels` from the structure.